### ISSUE TITLE
Bug: Topical embeddings have poor cross-topic discrimination

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#243]
## Problem

Topical embeddings evaluation shows **high intra-cluster coherence but poor cross-topic discrimination**.

### Evaluation Results
```
Cluster coherence (mean intra-cluster similarity):
  ✓ Esperanto Movement: 0.953
  ✓ Family: 0.970
  ✓ Politics: 0.950
  ✓ Animals: 0.935
  Average: 0.817 ✓ Good

Cross-topic similarity (should be LOW):
  ⚠️ Esperanto Movement ↔ Animals: 0.919
  ⚠️ Politics ↔ Family: 0.956
  ⚠️ Science ↔ Music: 0.657
  Average: 0.819 ⚠️ BAD (should be < 0.3)
```

### Symptoms
- Everything is similar to everything (0.9+ similarity across topics)
- High-frequency words dominate: `esperant` similar to `list, bibliotek, scienc` all at 0.99
- Model learned "words that appear often" not "words that co-occur in same topics"

### Root Cause Analysis
1. **Negative sampling too weak**: 3:1 ratio may not provide enough contrast
2. **TF-IDF filtering incomplete**: High-frequency words that appear in 50%+ docs still included
3. **Training objective**: Binary co-occurrence doesn't penalize cross-topic similarity enough

### Proposed Fixes

**Option A: Better data preparation**
- Increase negative ratio from 3:1 to 5:1 or 10:1
- Add hard negatives (words from different documents on same topic)
- Stricter TF-IDF filtering (max 5% of docs instead of 10%)

**Option B: Training objective change**
- Use triplet loss instead of binary classification
- Anchor: word, Positive: same-doc word, Negative: different-doc word
- This directly optimizes for discrimination

**Option C: Post-training calibration**
- Apply whitening/normalization to spread embeddings
- Use PCA to reduce correlation between dimensions

### Reproduction
```bash
./scripts/evaluate_topical_embeddings.sh
```

### Related
- Training script: `scripts/train_topical_embeddings.py`
- Data prep: `scripts/prepare_topical_training_data.py`
- Model: `models/topical_embeddings/best_model.pt` (9.8M params, 87.6% val accuracy)

The 87.6% validation accuracy is misleading - the model is good at the binary task but embeddings don't generalize to topic discrimination.
